mangaki/mangaki/views.py

```diff
@@ -807,6 +809,90 @@ def legal_mentions(request):
     return render(request, 'mangaki/legal.html')
 
 
+def fix_index(request):
+    suggestion_list = Suggestion.objects.select_related('work', 'user').prefetch_related(
+        'work__category', 'evidence_set__user').all().order_by('is_checked', '-date')
+
+    paginator = Paginator(suggestion_list, FIXES_PER_PAGE)
+    page = request.GET.get('page')
+
+    try:
+        suggestions = paginator.page(page)
+    except PageNotAnInteger:
+        suggestions = paginator.page(1)
+    except EmptyPage:
+        suggestions = paginator.page(paginator.num_pages)
+
+    context = {
+        'suggestions': suggestions
+    }
+
+    return render(request, 'fix/fix_index.html', context)
+
+
+def fix_suggestion(request, suggestion_id):
+    cluster_colors = {
+        'unprocessed': 'text-info',
+        'accepted': 'text-success',
+        'rejected': 'text-danger'
+    }
+
+    if not suggestion_id:
+        return redirect('fix-index')
+
+    suggestion = get_object_or_404(Suggestion.objects.select_related('work', 'user'), id=suggestion_id)
+
+    evidence = None
+    if request.user.is_authenticated:
+        evidence = Evidence.objects.filter(user=request.user, suggestion=suggestion).first()
+
+    cluster = WorkCluster.objects.filter(origin=suggestion_id).first()
+    cluster_works = cluster.works.all().prefetch_related('category') if cluster else None
+
+    # Get the previous suggestion, ie. more recent and of the same checked status
+    previous_suggestion = Suggestion.objects.filter(date__gt=suggestion.date,
+        is_checked=suggestion.is_checked).order_by('date').first()
+
+    # If there is no more recent suggestion, and was checked, just pick from not checked suggestions
+    if not previous_suggestion and suggestion.is_checked:
+        previous_suggestion = Suggestion.objects.filter(is_checked=False).order_by('date').first()
+
+    # Get the next suggestion, ie. less recent and of the same checked status
+    next_suggestion = Suggestion.objects.filter(date__lt=suggestion.date,
+        is_checked=suggestion.is_checked).order_by('-date').first()
+
+    # If there is no less recent suggestion, and wasn't checked, just pick from checked suggestions
+    if not next_suggestion and not suggestion.is_checked:
+        next_suggestion = Suggestion.objects.filter(is_checked=True).order_by('-date').first()
+
+    context = {
+        'suggestion': suggestion,
+        'related_cluster': cluster,
+        'cluster_works': cluster_works,
+        'cluster_colors': cluster_colors[cluster.status] if cluster else None,
+        'evidence': evidence,
+        'next_id': next_suggestion.id if next_suggestion else None,
+        'previous_id': previous_suggestion.id if previous_suggestion else None
+    }
+
+    return render(request, 'fix/fix_suggestion.html', context)
+
+
+def update_evidence(request):
+    agrees = request.POST.get('agrees')
```

Comment:
I updated and used a Django Form for updating or creating an `Evidence` object ;)
However but I'm not sure where to move the `update_evidence` method (if it has to be moved).

I also updated the form so that it can be used if the user doesn't have JavaScript enabled.